lib/src/models/collection_reference.dart

```diff
@@ -0,0 +1,179 @@
+import 'package:firedart/src/generated/google/firestore/v1/query.pb.dart';
+import 'package:firedart/src/generated/google/protobuf/wrappers.pb.dart';
+import 'package:firedart/src/generated/google/firestore/v1/document.pb.dart'
+    as fs;
+import 'package:firedart/src/models/document.dart';
+import 'package:firedart/src/models/document_reference.dart';
+import 'package:firedart/src/models/reference.dart';
+import 'package:firedart/src/repository/firestore_gateway.dart';
+import 'package:firedart/src/util/firestore_encoding.dart';
+
+class CollectionReference extends Reference {
+  StructuredQuery _structuredQuery;
+
+  CollectionReference(FirestoreGateway gateway, String path)
+      : super(gateway, path) {
+    _structuredQuery = StructuredQuery();
+    _structuredQuery.from
+        .add(StructuredQuery_CollectionSelector()..collectionId = id);
+    if (fullPath.split('/').length % 2 == 1) {
+      throw Exception('Path is not a collection: $path');
+    }
+  }
+
+  DocumentReference document(String id) {
+    return DocumentReference(gateway, '$path/$id');
+  }
+
+  /// Returns [CollectionReference] with additional filter on specified
+  /// [field]. [field] refers to a field in a document.
+  ///
+  /// The [field] may be a [String] consisting of a single field name
```

Comment:
```suggestion
  /// [fieldPath]. [fieldPath] refers to a field in a document.
  ///
  /// The [fieldPath] is a [String] consisting of a single field name
```
This will make documentation more interactable in IDE and when generated.